html/cross-origin-embedder-policy/reporting-to-endpoint.https.html

```diff
@@ -77,100 +77,82 @@
       reportsForReportOnly, url, iframe.src, 'reporting');
-
-    t.done();
-  } catch (e) {
-    t.step(() => { throw e });
-  }
 }, 'subresource CORP');
 
-async_test(async (t) => {
-  try {
-    const iframe = document.createElement('iframe');
-    t.add_cleanup(() => iframe.remove());
+promise_test(async t => {
+  const iframe = document.createElement('iframe');
+  t.add_cleanup(() => iframe.remove());
 
-    iframe.src = `resources/reporting-empty-frame.html`
-    document.body.appendChild(iframe);
-    await new Promise(resolve => {
-      iframe.addEventListener('load', resolve, {once: true});
-    });
+  iframe.src = `resources/reporting-empty-frame.html`
+  document.body.appendChild(iframe);
+  await new Promise(resolve => {
+    iframe.addEventListener('load', resolve, {once: true});
+  });
 
-    const w = iframe.contentWindow;
+  const w = iframe.contentWindow;
 
-    function attachFrame(url) {
-      const frame = w.document.createElement('iframe');
-      frame.src = url;
-      w.document.body.appendChild(frame);
-    }
+  function attachFrame(url) {
+    const frame = w.document.createElement('iframe');
+    frame.src = url;
+    w.document.body.appendChild(frame);
+  }
 
-    const url = `${REMOTE_ORIGIN}/common/blank.html?${token()}`;
-    // The nested frame comes from cross-origin and doesn't have a CORP
-    // header, so it is blocked.
-    attachFrame(url);
+  const url = `${REMOTE_ORIGIN}/common/blank.html?${token()}`;
+  // The nested frame comes from cross-origin and doesn't have a CORP
+  // header, so it is blocked.
+  attachFrame(url);
 
-    // Wait 3 seconds for reports to settle.
-    await wait(3000);
+  // Wait 3 seconds for reports to settle.
+  await wait(3000);
 
-    checkCorpReportExistence(reports, url, iframe.src, 'enforce');
-    checkCorpReportExistence(
+  checkCorpReportExistence(reports, url, iframe.src,'enforce');
+  checkCorpReportExistence(
       reportsForReportOnly, url, iframe.src, 'reporting');
-
-    t.done();
-  } catch (e) {
-    t.step(() => { throw e });
-  }
 }, 'navigation CORP');
 
-async_test(async (t) => {
-  try {
-    const iframe = document.createElement('iframe');
-    t.add_cleanup(() => iframe.remove());
+promise_test(async (t) => {
+  const iframe = document.createElement('iframe');
+  t.add_cleanup(() => iframe.remove());
 
-    iframe.src = 'resources/reporting-empty-frame.html';
-    const targetUrl = `/common/blank.html?${token()}`;
-    iframe.addEventListener('load', () => {
-      const nested = iframe.contentDocument.createElement('iframe');
-      nested.src = targetUrl;
-      // |nested| doesn't have COEP whereas |iframe| has, so it is blocked.
-      iframe.contentDocument.body.appendChild(nested);
-    }, {once: true});
+  iframe.src = 'resources/reporting-empty-frame.html';
+  const targetUrl = `/common/blank.html?${token()}`;
+  iframe.addEventListener('load', () => {
+    const nested = iframe.contentDocument.createElement('iframe');
```

Comment:
If there's any chance this code could throw it should be wrapped in `t.step_func`. Looks like it's using pretty old APIs, so I guess it's OK to omit.